docs/tutorials/intro.md

```diff
@@ -223,12 +183,74 @@ fn main() {
+
+If you have ICU4X data on the file system in a JSON format, it can be loaded via `FsDataProvider`:
+
+```toml
+[dependencies]
+icu = "0.5"
+icu_provider_fs = {version = "0.5" , features = ["deserialize_json"]}
+```
+
+```rs
+use icu_provider_fs::FsDataProvider;
+
+fn main() {
+    let _provider = FsDataProvider::try_new("/path/to/data")
+       .expect("Failed to initialize Data Provider.");
+}
+```
+
+The ICU4X repository has test data checked in tree in `provider/testdata/data/json`, however it is recommended one generate data on their own as described in the [next section](#generating data). Under the hood, `icu_testdata::get_provider()` is simply loading this data.
+
+## Generating data
+
+For production usage, it is better to generate your own data that is filtered to suit your needs.
+
+We're going to use [JSON CLDR](https://github.com/unicode-cldr/cldr-json) as our source data. JSON CLDR is an export of [CLDR data](http://cldr.unicode.org/index/downloads) into JSON maintained by Unicode.
+
+We are also going to use Unicode property data shipped as a zip file in the ICU4C release.
+
+The `datagen` component has a binary application which will fetch the CLDR data and generate ICU4X data out of it.
+
+```
+cd ~/projects/icu
+wget https://github.com/unicode-org/icu/releases/download/release-70-1/icuexportdata_uprops_full.zip
+unzip icuexportdata_uprops_full.zip
+git clone https://github.com/unicode-org/icu4x
+cd icu4x
+git checkout icu@0.5.0
+cargo run --bin icu4x-datagen -- \
+    --cldr-tag 40.0.0 \
+    --uprops-root ../icuexportdata_uprops_full/small \
+    --out ~/projects/icu/icu4x-data \
+    --all-keys --all-locales
+```
+
+The last command is a bit dense, so let's dissect it.
+
+* First, we call `cargo run` which runs a binary in the crate
+* We tell it that the binary is named `icu4x-datagen`
+* Then we use `--` to separate arguments to `cargo` from arguments to our app
+* Then we pass `--cldr-tag` which informs the program which CLDR version to use
+* Then we pass `--uprops-root` which informs the program where to get the Unicode property data
+* Then we pass `--out` directory which is where the generated ICU4X data will be stored
+* Finally, we set `--all-keys` which specify that we want to export all keys available
+
+After that step, it should be possible to navigate to `~/projects/icu/icu4x-data` and there should be a `manifest.json` file, and directories with data.
+
+*Notice:* In this tutorial we export data as compact `JSON` which provides decent performance and readable data files. There are other formats and options for formatting of the data available. Please consult `cargo run --bin icu4x-datagen -- --help` for details.
+*Notice:* In particular, in production, the `postcard` format will yield better performance results.
+*Notice:* For offline or unconventional use, the user can also pass `--cldr-core` and `--cldr-dates` paths to local clones of the repositories instead of `--cldr-tag`.
```

Comment:
hm? Do you mean the user can pass a custom CLDR directory with `--cldr-root`?

Reply:
Yeah that's legacy stuff that was already present, fixed.

Reply:
If we're calling `--cldr-root` "unconventional", maybe we should make the `"download"` feature default? @sffc